### ISSUE TITLE
Tank entity — third person view, perspective, and controls

### ISSUE THREAD

**Issue by chase-moskal** [ChaseMoskal/Nanoshooter#22]
Aim controls via mouse:
- 3rd person camera angle.
- Mouse movement freely swivels the camera in a 360 degree orb around the tank.
- Exact center of aim projects a target aimpoint.
- Gun turret and cannon actively swivels and points exactly toward the target aimpoint.

Enhanced movement controls:
- WASD movement controls are relative to the camera angle (allowing strafing).
- Shift-key provides a "sprint" boost that only works in the forward direction.
- Moving diagonally isn't faster than moving in other directions.

The above criteria was mostly explained from the perspective of a player-controlled tank, though we need NPC tanks to exhibit the same movement behavior that players do. This means that we need to abstract the movement controls in a way that NPC tanks can equally leverage these controls, by placing their own aimpoint in an arbitrary location that has nothing to do with mouse input, and can choose their desired movement vector in a way that has nothing to do with keyboard input.. you get the idea.

This ticket probably doesn't need to include a full tank physics overhaul, which should probably be a separate ticket. This is more specifically about the way that the controls are applied, not so much the details of the physical model onto which they are applied. I like to think these can be done separately.

**Comment by chase-moskal** [comment]
Probably want to wait on #18 to close before starting this work.
